interceptors/selector/selector.go

```diff
@@ -0,0 +1,58 @@
+package selector
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+)
+
+type MatchFunc func(ctx context.Context, fullMethod string) bool
+
+type options struct {
+	match MatchFunc
+}
+
+type Option func(o *options)
+
+// WithMatch ...
+func WithMatch(fn MatchFunc) Option {
+	return func(o *options) {
+		o.match = fn
+	}
+}
+
+func evaluateServerOpt(opts []Option) *options {
+	optCopy := &options{}
+	optCopy.match = AlwaysMatch
+	for _, o := range opts {
+		o(optCopy)
+	}
+	return optCopy
+}
+
+// AlwaysMatch is the default implementation, always selected
+func AlwaysMatch(_ context.Context, _ string) bool {
+	return true
+}
+
+// UnaryServerInterceptor returns a new unary server interceptors that performs per-request selector.
+func UnaryServerInterceptor(interceptors grpc.UnaryServerInterceptor, opts ...Option) grpc.UnaryServerInterceptor {
```

Comment:
I'm leaning towards removing `MatchFunc` as an option and making it a required parameter. This solves two problems:

1. If you accidentally use the selector without a match func, you gain no utility from it.
2. If you accidentally specify two match func options, you might be confused about which to use.

That simplifies things generally, no need for the `AlwaysMatch` function, etc.

Reply:
Yes, I readjusted it, can you help me in seeing what is wrong?